Mathlib/RingTheory/TensorProduct/MvPolynomial.lean

```diff
@@ -186,6 +186,31 @@ noncomputable def scalarRTensorAlgEquiv :
     MvPolynomial σ R ⊗[R] N ≃ₐ[R] MvPolynomial σ N :=
   rTensorAlgEquiv.trans (mapAlgEquiv σ (Algebra.TensorProduct.lid R N))
 
+variable (A : Type*) [CommRing A] [Algebra R A]
+
+/-- Tensoring `MvPolynomial σ R` on the left by an `R`-algebra `A` is algebraically
+equivalent to `M̀vPolynomial σ A`. -/
+noncomputable def algebraTensorAlgEquiv :
+    A ⊗[R] MvPolynomial σ R ≃ₐ[A] MvPolynomial σ A := AlgEquiv.ofAlgHom
+  (Algebra.TensorProduct.lift
+    (Algebra.ofId A (MvPolynomial σ A))
+    (MvPolynomial.mapAlgHom <| Algebra.ofId R A) (fun _ _ ↦ Commute.all _ _))
+  (aeval (fun s ↦ 1 ⊗ₜ X s))
+  (by ext s; simp)
+  (by ext s; simp)
+
+@[simp]
+lemma algebraTensorAlgEquiv_apply (a : A) (s : σ) :
```

Comment:
This should be called `algebraTensorAlgEquiv_tmul_X`, and I think we should have `algebraTensorAlgEquiv_tmul` as the simp lemma (`algebraTensorAlgEquiv (R := R) A (a ⊗ₜ p) = a • p.map (algebraMap R A)`)

Reply:
Should I keep the `tmul_X` version? With the new lemma it is proven by `simp`.

Reply:
Hmmm maybe not. Sometimes it still useful to restate lemma provable by simp for rewriting, but this lemma is merely two rewrites.